### DISCUSSION TITLE
documentation for just using straight web-components?

### DISCUSSION THREAD

**Discussion by acroyear** [fwouts/previewjs#2602] [Ideas]
The examples/setup are all for the big frameworks. I'm at the other end, where most of my work is just writing plain web-components (which i might later utilize in one of those frameworks). I'm generally on a pin to sticking to zero-dependencies for some of the work i need to do.

So basically I'd be interested in documentation and setup that supports just making and demoing vanilla-js web-components.

**Comment by fwouts** [comment]
That's a good idea, there just hasn't been much demand for it. I've never written web-components before, so I'm just not sure what patterns to look for.

Are they always implemented as subclasses of HTMLElement, like `class HelloWorld extends HTMLElement {`?

If you have an example repo that we could use to test this out, it'd be helpful too.

**Reply by acroyear** [reply]
sorry, after this it became a VERY busy spring. i've unread emails (I glance at subjects and ignore a hell of a lot) going back here to April. It wouldn't be too hard to take my home page repo and copy it up to github (I run on bitbucket).
